src/schema/users.ts

```diff
@@ -1176,6 +1193,68 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
 
       return streak;
     },
+    streakRecover: async (
+      _,
+      __,
+      ctx: AuthContext,
+    ): Promise<StreakRecoverQueryResult> => {
+      const cantRecoverResult: StreakRecoverQueryResult = {
+        canDo: false,
+        cost: 0,
+        oldStreakLength: 0,
+      };
+
+      const { userId } = ctx;
+
+      const key = generateStorageKey(
+        StorageTopic.Streak,
+        StorageKey.Reset,
+        userId,
+      );
+
+      const oldStreakLength = Number(await getRedisObject(key));
+      const userDontHaveOldStreak =
+        !!oldStreakLength && !isNumber(oldStreakLength);
+      if (userDontHaveOldStreak) {
+        return cantRecoverResult;
+      }
+
+      const streak = await ctx.con.getRepository(UserStreak).findOneBy({
+        userId,
+      });
+
+      if (!streak) {
+        return cantRecoverResult;
+      }
+
+      const { currentStreak } = streak;
+      const streakIsTooLongToRecover = currentStreak > 1;
+      if (streakIsTooLongToRecover) {
+        await ctx.con.getRepository(Alerts).update(
+          {
+            userId,
+          },
+          {
+            showRecoverStreak: false,
+          },
+        );
+        return cantRecoverResult;
+      }
+
+      const recoverCount = await ctx.con
```

Comment:
Was there not also some rule to say don't recover if recovered yesterday? (or this above query handles it?)

Reply:
@rebelchris We are handling the checks in the query based on the Redis cache of the user's lost streak. Essentially, from PR #2151 onward, when a user loses their streak, we store a 24-hour value, so we only check for this.

If you recovered yesterday and didn't read anything until today (causing you to lose the streak again), on the streak update cron, the new value will be set in Redis, and you can recover again. Does that logic make sense?

cc @sshanzel

Reply:
Ah got it, yes makes sense :) 🙏